### PR TITLE
Add executable tests and use mapbox-sdk-js

### PR DESCRIPTION
**Pull request by jakepruitt**
- refs #10

### CODE REVIEW

**Comment by jakepruitt** [comment]
@tmcw Is there a specific reason you wanted node v0.8 support? It looks like update-notifier [doesn't support anything less than node v0.10](https://github.com/yeoman/update-notifier/blob/master/package.json#L13), so travis is failing the v0.8 builds.

Should we get rid of the node v0.8 requirement or kill off update-notifier?

**Comment by rclark** [comment]
I think the command names are pretty confusing because there are now two kinds of resources we're interacting with (datasets, and features within datasets). What we present this as one command-line utility, and differentiate sub-commands, like:

```
# list datasets in your account
mapbox-data list-datasets

# create a dataset
mapbox-data create-dataset [name] [description]

# mess with features
mapbox-data insert-features <datasetid> [filename]
mapbox-data get-features <datasetid>

# replace
mapbox-data replace-dataset <datasetid> [filename]
```

**Comment by rclark** [comment]
:100: wow I love the refactor! My only concern is that all the functions are all funneled through the same `handler` callback - this might prove to be a chokepoint. As an experiment let me try and add a `dump-dataset` function that exports the entire dataset as a geojson featurecollection.

File: mapbox-data [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    'delete-feature': client.deleteFeature.bind(client, argv._[2], argv._[1]),
+    'delete-dataset': client.deleteDataset.bind(client, argv._[1]),
+    'get-feature': client.readFeature.bind(client, argv._[2], argv._[1]),
+    'get-dataset': client.readDataset.bind(client, argv._[1]),
+};
+
+cmds[argv._[0]](handler);
+
+function handler(err, result) {
+    if (err || result === undefined) return console.error(err);
```
Comment (rclark) [inline mapbox-data]: In a CLI environment, its nice to make sure that the process exits with a non-zero exit code in the event of a failure. You can do that one of two ways;

``` js
// if you want stack traces
if (err) throw err;
// ... or if you don't ...
if (err) {
  console.error(err.message || 'An unexpected error occurred');
  process.exit(1);
}
```

I would also suggest you differentiate these conditions:

``` js
if (err) { ... return the information in the error object ... }
else if (!result) { ... inform the user that something unexpected happened ... }
```

**Comment by tmcw** [comment]
:+1: LGTM

**Pull request merged**